config/config-api/src/test/java/com/thoughtworks/go/config/CruiseConfigTestBase.java

```diff
@@ -811,24 +814,27 @@ public void getAllUniquePostCommitSchedulableMaterials_shouldReturnMaterialsWith
     }
 
     @Test
-    public void getAllUniquePostCommitSchedulableMaterials_shouldReturnMaterialsWithAutoUpdateFalseAndConfigRepos() {
-        GitMaterialConfig gitAutoMaterial = MaterialConfigsMother.gitMaterialConfig("url");
-        PipelineConfig pipelineAuto = pipelineConfig("pipelineAuto", new MaterialConfigs(gitAutoMaterial));
-        GitMaterialConfig gitNonAutoMaterial = git(new UrlArgument("other-url"), null, null, "master", "dest", false, null, false, null, new CaseInsensitiveString("git"), false);
-        PipelineConfig pipelineTriggerable = pipelineConfig("pipelineTriggerable", new MaterialConfigs(gitNonAutoMaterial));
+    public void getAllUniquePostCommitSchedulableMaterials_shouldReturnMaterialsAndConfigReposWithAutoUpdateFalse() {
+        GitMaterialConfig gitMaterialAuto = MaterialConfigsMother.gitMaterialConfig("url");
+        PipelineConfig pipelineAuto = pipelineConfig("pipelineAuto", new MaterialConfigs(gitMaterialAuto));
+        GitMaterialConfig gitMaterialManual = git(new UrlArgument("other-url"), null, null, "master", "dest", false, null, false, null, new CaseInsensitiveString("git"), false);
+        PipelineConfig pipelineTriggerable = pipelineConfig("pipelineTriggerable", new MaterialConfigs(gitMaterialManual));
         PipelineConfigs defaultGroup = createGroup("defaultGroup", pipelineAuto, pipelineTriggerable);
 
         cruiseConfig = new BasicCruiseConfig(defaultGroup);
         ConfigReposConfig reposConfig = new ConfigReposConfig();
-        GitMaterialConfig configRepoMaterial = git("http://git");
-        reposConfig.add(ConfigRepoConfig.createConfigRepoConfig(configRepoMaterial, "myplug", "id"));
+        GitMaterialConfig configRepoMaterialAutoUpdate = git("http://git");
+        GitMaterialConfig configRepoMaterialManual = tap(git("http://git2"), g -> g.setAutoUpdate(false));
+        reposConfig.add(ConfigRepoConfig.createConfigRepoConfig(configRepoMaterialAutoUpdate, "myplug", "exclude"));
+        reposConfig.add(ConfigRepoConfig.createConfigRepoConfig(configRepoMaterialManual, "myplug", "include"));
         cruiseConfig.setConfigRepos(reposConfig);
 
 
         Set<MaterialConfig> materials = cruiseConfig.getAllUniquePostCommitSchedulableMaterials();
         assertThat(materials.size(), is(2));
-        assertThat(materials, hasItem(gitNonAutoMaterial));
-        assertThat(materials, hasItem(configRepoMaterial));
+        assertThat(materials, hasItem(gitMaterialManual));
+        assertThat(materials, hasItem(configRepoMaterialManual));
+        assertThat(materials, not(hasItem(configRepoMaterialAutoUpdate)));
```

Comment:
Changing this test to reflect that only materials and configrepos with `auto_update = false` will be considered.